rotary_utils/rotate.py

```diff
@@ -0,0 +1,439 @@
+                                     rotate_values=rotate_values_dict, sequence_names=sequence_names,
+                                     strip_descriptions=args.strip_descriptions)
+
+    else:
+        raise RuntimeError('Could not find any rotate commands to execute.')
+
+    # Save the rotation report if desired
+    if args.output_report is not None:
+        logger.debug(f'Writing rotation report to "{args.output_report}"')
+        report.to_csv(args.output_report, sep='\t', index=False)
+
+    logger.debug(os.path.basename(sys.argv[0]) + ': done.')
+
+
+def check_rotate_args(args):
+    """
+    Checks that the arguments supplied to the rotate module are non-conflicting. Errors if a conflict is found.
+
+    :param args: Arguments parsed by subparse_cli()
+    """
+
+    """
+    As described in the README, only one of the following arguments is allowed to be set for a run:
+    
+    args.midpoint
+    args.rotate_position
+    args.rotate_fraction
+    args.rotate_position_table
+    args.rotate_fraction_table
+    """
+    defined_arguments = 0
+    for argument in [args.rotate_position, args.rotate_fraction, args.rotate_position_table,
+                     args.rotate_fraction_table]:
+        if argument is not None:
+            defined_arguments = defined_arguments + 1
+
+    # Special case for boolean midpoint setting
+    if args.midpoint is True:
+        defined_arguments = defined_arguments + 1
+
+    try:
+        if defined_arguments > 1:
+            raise RuntimeError('More than one of -m, -p, -P, -t, and -T were specified in the command line, but only '
+                               'one of these can be set for a rotate run.')
+        elif defined_arguments == 0:
+            raise RuntimeError('None of -m, -p, -P, -t, and -T were specified in the command line, but you must select '
+                               'one of these to perform a rotate run.')
+        elif defined_arguments != 1:
+            raise RuntimeError('Ran into an issue processing the -m, -p, -P, -t, and -T arguments in the CLI.')
+    except Exception as error:
+        logger.error(str(error))
+        raise
+
+
+def parse_rotate_value_table(rotate_table_filepath: str, rotate_type: str):
+    """
+    Parses a tab-separated table as defined in the CLI containing rotation information.
+
+    :param rotate_table_filepath: path to the tab-separated table with sequence rotation info
+    :param rotate_type: either 'position' or 'fraction'. Runs a check to make sure column names match the specified
+                        rotation type.
+    :return: dictionary of sequence names (keys) and rotation values (values)
+    """
+
+    if rotate_type == 'position':
+        expected_column_names = ['sequence_id', 'rotate_bp']
+    elif rotate_type == 'fraction':
+        expected_column_names = ['sequence_id', 'rotate_fraction']
+    else:
+        error = ValueError(f'Expected "position" or "fraction", but got "{rotate_type}".')
```

Comment:
@LeeBergstrand  Quick question about `ValueError`: here, I raise a `ValueError` if the user provides an unexpected value for `rotate_type`. However, I don't check the actual type provided. If the user provides a `dict` or `list` or something, I assume it would be more specific/accurate to raise a [`TypeError`](https://docs.python.org/3/library/exceptions.html#TypeError). However, I'm a bit hesitant to add more code to distinguish between ValueErrors and TypeErrors in cases like this one, because the function is used internally and is not really intended to be exposed to the user. Seems a bit unnecessarily bulky to me to add another layer of exception handling. What are your thoughts? Thanks!

Reply:
@jmtsuji I would leave it as `ValueError` and not pursue the additional code. If the wrong type is passed inadvertently, you will see it in the error message as `{rotate_type}` text and can pick it up from their. We don't need to be that specific. We just need to raise the most common error.

Reply:
Sounds good -- will leave these as `ValueError`, as-is.